sphinxcontrib/dotnetdomain.py

```diff
@@ -181,44 +181,48 @@ def handle_signature(self, sig_input, signode):
         return sig.full_name(), sig.prefix
 
     def add_target_and_index(self, name_obj, sig, signode):
-        # TODO wtf does this do?
-        objectname = self.options.get(
-            'object', self.env.ref_context.get('dn:object'))
-        fullname = name_obj[0]
-        if fullname not in self.state.document.ids:
-            signode['names'].append(fullname)
-            signode['ids'].append(fullname.replace('$', '_S_'))
+        '''Add objects to the domain list of objects
+
+        This uses the directive short name along with the full object name to
+        create objects and nodes that are type and name unique.
+        '''
+        obj_name = self.options.get('object',
```

Comment:
Is there a reason we're getting this obj_name here, but not using it except to pass into the next function? Why not just have this logic in get_index_text()?